src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js

```diff
@@ -27,34 +28,36 @@ const CustomLessonsDisplay = ({ firebase }) => {
   }
 
   useEffect(() => {
-    // Get custom lessons made by admin
-    firebase.getAdminCustomLessons(ADMIN_ACCOUNT).then(lesson => {
-      setAdminLessons(lesson);
-    });
-  }, [firebase]);
+    setTimeout(() => {
+      // Get custom lessons made by admin
+      firebase.getAdminCustomLessons(ADMIN_ACCOUNT).then(lesson => {
+        setAdminLessons(lesson);
+      });
+    }, 50); // Timeout for firebase to update and display updated lessons properly
+  }, [history.length]); // Updates lessons when redirected to page from CreateAssignment
```

Comment:
hmm was this not updating when there was an update in lessons? seems kind of sus to set a timeout only based off of router history length

Reply:
Yeah this was a slight workaround to make sure the page refreshes properly after redirection (which it normally doesn't), the simplest way to check if there was a redirect is by tracking history.length as it won't check for firebase changes (thus pushing to history above). The setTimeout() though is tough because it varies by computer wifi how fast firebase updates, not sure the optimal solution there

Reply:
ngl i'm not a very big fan of this- i'm not sure how applicable this is, but you could look at how jason updates the login screen when someone signs in